File: Gauntlet of the Hero - GOTH/Assets/_Scripts/Models/Backend/BattleSessionDto.cs
using System;
using Newtonsoft.Json;

namespace GauntletOfTheHero.Backend.Models
{
    [System.Serializable]
    public class BattleSessionDto
    {
        [JsonProperty("sessionId")]
        public Guid sessionId;

        [JsonProperty("hero")]
        public HeroDto hero;

        [JsonProperty("monster")]
        public MonsterDto monster;

        [JsonProperty("battleState")]
        public BattleStateDto battleState;
    }
}

File: nordeus_challenge_backend_spring_boot/src/main/java/com/mmileticc/nordeuschallenge/model/dto/BattleSessionDTO.java
package com.mmileticc.nordeuschallenge.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BattleSessionDTO {

    private UUID sessionId;
    private HeroDTO hero;
    private MonsterDTO monster;
    private BattleStateDTO battleState;
}
